libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -15,6 +15,8 @@
 
 extern AP_HAL::HAL& hal;
 
+Compass Compass::frontend;
+
```

Comment:
why do we need this as opposed to make the whole Compass class a singleton and not change anything else, like was done in AP_InertialSensor?